### PR TITLE
Improve tickets feature

### PR DESCRIPTION
**Pull request by elguitarraverde**
- Search for a ticket
- Create a ticket_message table
- Send email to the customer when the ticket is solved, or updated status

[Tickets issue](https://github.com/Roskus/hammercrm/issues/25)

### CODE REVIEW

**Comment by gnovaro** [comment]
We appreciate your amazing work! i let you a few suggestions but for me looks great so far so good

**Review by gnovaro — CHANGES_REQUESTED** [review]

File: app/Models/Ticket.php [hunk 1]
```diff
@@ -70,4 +73,54 @@ public function types(): array
...
+    }
+
+    protected static function booted()
+    {
+        static::updated(function ($ticket) {
+            if ($ticket->wasChanged('status')) {
+                if ($ticket->status == 'closed') {
+                    Mail::to($ticket->customer->email)->send(new GenericEmail(
+                        $ticket->company,
+                        'The ticket has been closed.',
```
[the remark this answers is not in the document]
Comment (elguitarraverde) [reply app/Models/Ticket.php]: ok. done!

File: app/Models/Ticket.php [hunk 2]
```diff
@@ -70,4 +73,54 @@ public function types(): array
...
+                        ['body' => 'The ticket has been closed.']
```
[the remark this answers is not in the document]
Comment (elguitarraverde) [reply app/Models/Ticket.php]: the GenericEmail class already has a view referenced: 'mail.generic'.

https://github.com/Roskus/hammercrm/blob/b19f2afb4d8ba3cf57365b931bd8bc5156f9f4d8/app/Mail/GenericEmail.php#L31-L37

File: database/migrations/2023_02_05_170842_create_messages_table.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('messages', function (Blueprint $table) {
```
[the remark this answers is not in the document]
Comment (elguitarraverde) [reply database/migrations/2023_02_05_170842_create_messages_table.php]: ok. Done!

File: database/seeders/DatabaseSeeder.php [hunk 4]
```diff
@@ -23,5 +23,6 @@ public function run()
         $this->call(UserSeeder::class);
         $this->call(IndustrySeeder::class);
         $this->call(BankSeeder::class);
+        $this->call(TicketSeeder::class);
```
[the remark this answers is not in the document]
Comment (elguitarraverde) [reply database/seeders/DatabaseSeeder.php]: ok. Done!

File: resources/views/mail/generic.blade.php [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
 <head>
-    <title>Laravel Mail Queue Tutorial</title>
+    <title>ROSKUS</title>
```
[the remark this answers is not in the document]
Comment (elguitarraverde) [reply resources/views/mail/generic.blade.php:4]: ok. We can implement an email template like this and just change the body.

https://github.com/leemunroe/responsive-html-email-template

File: app/Models/Ticket.php [hunk 6]
```diff
@@ -10,9 +10,15 @@
...
 use Illuminate\Database\Eloquent\Relations\HasOne;
 use Illuminate\Support\Facades\Mail;
+use Illuminate\Support\Str;
 use OpenApi\Attributes as OAT;
 
 #[OAT\Schema(schema: 'Ticket', required: ['title', 'description'], type: 'object')]
+#[OAT\Property(property: 'id', type: 'int', example: 1)]
+#[OAT\Property(property: 'title', type: 'string', example: 'An issue occur with the order #3')]
+#[OAT\Property(property: 'description', type: 'string', example: 'My order dosen\'t include one of the products we request')]
+#[OAT\Property(property: 'customer_id', type: 'int', example: 1)]
```
Comment (elguitarraverde) [inline app/Models/Ticket.php:20]: he tenido que poner esto así y eliminar las propiedades porque me daba error al usar el modelo en el evento updated.

File: app/Models/Ticket.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (elguitarraverde) [reply app/Models/Ticket.php]: https://github.com/Roskus/hammercrm/pull/81/commits/7b01334a823f934ef2fdf9c64e21883c63ac30ee#r1097853519

File: resources/views/mail/generic.blade.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (gnovaro) [reply resources/views/mail/generic.blade.php:4]: I like the idea but without table emails are really old.. browser now support better HTML emails, some people still maintain old backward compatibility....

File: app/Models/Ticket.php (hunk 6 above)
Comment (gnovaro) [reply app/Models/Ticket.php:20]: I had a similiar error we can review this better in the furute for now i will approve an merge the change.

**Review by gnovaro — APPROVED** [review]
Great job! thank you!

**Pull request merged**